test/test-project/gulpfile.js

```diff
@@ -48,5 +68,21 @@ gulp.task('test1', () => {
     // TODO(justinfagnani): allow filtering of files before writing
     .pipe(gulp.dest('build/unbundled'));
 
-  return mergeStream(bundled, unbundled);
+  return Promise.all([waitFor(bundled), waitFor(unbundled)]).then(() => {
```

Comment:
for the demo gulpfile, what do you think about using `.on('end')` instead of a Promise helper function?